tests/e2e/koperator.go

```diff
@@ -0,0 +1,79 @@
+// Copyright © 2023 Cisco Systems, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"emperror.dev/errors"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	. "github.com/onsi/ginkgo/v2"
+)
+
+// applyKoperatorSampleResource deploys the specified sample resource (config/samples).
+// The full path of the manifest also can be specified.
+// It supports different versions that can be specified with the koperatorVersion parameter.
+func applyKoperatorSampleResource(kubectlOptions k8s.KubectlOptions, koperatorVersion Version, sampleFile string) error {
```

Comment:
**Question**: which part of this function is not covered by `k8sResourcesFromManifestPaths()`?

We do the split for CRDs anyway and we apply local files for local version while URL resources for remote version.

IMO we should be able to adapt that to this use case.

I feel like this is a heavily specialized function I would avoid if possible, sorry I don't have time to deeply compare now.

Reply:
Sure it can be adapted I think but at the same time this looks a lot nicer in my opinion as this doesn't require full paths to be given. I'll change it anyway.